### ISSUE TITLE
Rewrite `emu.py` in pytorch?

### ISSUE THREAD

**Issue by htjb** [handley-lab/stemu#3]
@williamjameshandley Curious what we think about doing this? I have been playing with `pytorch` recently, and I am happy to give it a go.

`pytorch` seems to have gained a bit more traction in the computer science literature recently, and I think it is a bit more stable than tensorflow.

**Comment by williamjameshandley** [comment]
Sounds like a good idea -- ideally we would have several side-by-side to test against, with the same scikit-learn/pandas machinery to organise. Have a think about the best way to refactor the code such that the 'model' part is separated out so that it doesn't have to be tensorflow.

@yallup may have some opinions here too.

**Comment by yallup** [comment]
I believe this is the goal of Keras3 already, a higher level api that can leverage tf, jax, torch backends, if you wanted backend agnosticism that is probably the way to go. Personally I would suggest just leaning into jax and being done with it, I can submit a PR with a basic flax (jax) setup as I have some similar existing projects to this

**Comment by htjb** [comment]
@williamjameshandley yes we could probably write a backend agnostic base class for the model which the user can build on and have example tfmodel, torchmodel, jaxmodel classes.

@yallup we can recommend jax over tensorflow/pytorch. Maybe I can PR a base class and modify the existing keras code into a keras/tf class. Then we can add a jax class?
